### PR TITLE
Notify GDPP on Settings Change

### LINKED ISSUE #21353: Expose the Q-summation functionality on the Settings tab
As described in #21167, this issue is to Expose the Q-summation functionality on the Settings tab via via SummationType and ReductionType drop downs as on the RROA dialog.

### PR DESCRIPTION
**Pull request by edwardb96**
This PR removes the need for the m_settingsChanged flag in the GenericDataProcessorPresenter by making the Reflectometry GUI actively notify it when the settings are changed, this eliminates the need for the `areSettingsChanged` routine and should make it easier to switch the parallelisation model in the `GenericDataProcessorPresenter`.

Additionally it includes the changes from #21364 with the test + compilation failures fixed.
> Exposes the `ReductionType` and `SummationType` options of `ReflectometryReductionOneAuto` on the > Reflectometry GUI settings tab via two comboboxes.

**To test:**
1. Ensure that the Reflectometry Settings tab still correctly passes the settings to the reduction algorithms.
2. Ensure that the SANS v2 GUI continued to function as expected.

Fixes #21353. 

**Release Notes** 
*Does not need to be in the release notes.*

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####

- [ ] Is the code of an acceptable quality?
- [ ] Does the code conform to the [coding standards](http://www.mantidproject.org/Coding_Standards)?
- [ ] Are the unit tests small and test the class in isolation?
- [ ] If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- [ ] Do changes function as described? Add comments below that describe the tests performed?
- [ ] Do the changes handle unexpected situations, e.g. bad input?
- [ ] Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: qt/widgets/common/src/DataProcessorUI/TwoLevelTreeManager.cpp [hunk 1]
```diff
@@ -111,6 +111,12 @@ std::vector<Command_uptr> TwoLevelTreeManager::publishCommands() {
   return commands;
 }
 
+void TwoLevelTreeManager::invalidateAllProcessed() {
+  for (auto i = 0; i < m_model->rowCount(); i++) {
+    setProcessed(false, i);
+  }
```
Comment (gemmaguest) [inline qt/widgets/common/src/DataProcessorUI/TwoLevelTreeManager.cpp]: I think this needs to loop through the group/row hierarchy in a similar way to [here](https://github.com/mantidproject/mantid/blob/master/qt/widgets/common/src/DataProcessorUI/QTwoLevelTreeModel.cpp#L221)

**Review by gemmaguest — CHANGES_REQUESTED** [review]
The changes look good, but I think there is an error in setting the processed state of rows in the two level tree - see inline comments

**Comment by gemmaguest** [comment]
The changes look good now and the reflectometry GUI works as expected. Rows are reprocessed when settings/data change but not otherwise.

There seems to be a problem on the SANS GUI - if you load the maskfile and batch file from the sample data and tick `PlotResults`, then the workspaces are created ok but the plot window does not show any graphs.

**Comment by gemmaguest** [comment]
The SANS GUI seems to be working fine now. Changes to the settings on the runs tab or the settings tab seem to be picked up ok. Therefore all looks good to me :+1:

**Review by gemmaguest — APPROVED** [review]

**Comment by martyngigg** [comment]
@edwardb96 It looks like there are some conflicts after #21343 was merged.

**Pull request merged**